### PR TITLE
[#30][#32] Setting up CI for the project

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/nimblehq/flutter_templates/issues/30
https://github.com/nimblehq/flutter_templates/issues/32

## What happened 👀

We need the CI to validate our integration continuously.

We need to add a `flutter analyze` step to CI workflow to ensure the code quality.

- Clean up all project lint warnings.
- Update CI workflow to have flutter analyze step in Github workflow.

## Insight 📝

- Add a `test` workflow to check code formatting, static code analyzing, and unit testing.

## Proof Of Work 📹

- A `test` workflow should be run properly on Github Action.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      macro: no
+
+comment:
+  layout: "reach,diff,flags,files,tree"
+  behavior: default
+  require_changes: no
+
+ignore:
+  - "lib/di"
+  - "lib/gen"
```
Comment (sleepylee) [inline codecov.yml:26]: I think you will need to ignore the same amount of files in .gitignore

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Sample CodeCov config for private repo
+codecov:
+  bot: luongvo
```
Comment (sleepylee) [inline codecov.yml:3]: possibly not need this

Comment (luongvo) [reply codecov.yml:3]: I put a comment above ☝️ , we need it for the private repository. We create this template for any project, right? Should we go to resolve issues later with private repo integration without this config? 🤓

Comment (sleepylee) [reply codecov.yml:3]: @luongvo I'm not sure if it should be the case for every project 🤔 . Check #21 I think I mentioned clearly that it should be scoped for this template_project only - to make sure that our code meets the standard before being used by someone else. Now that we are having:
- Flagging system for add-on features
- Incoming `example/` codebase

I think we need to plan strategically what will happen next. I.e: if the client doesn't want to use this service, we will have to remove this config. Then should it be flag as an optional feature? etc...

Comment (manh-t) [reply codecov.yml:3]: @luongvo I think we should let this config as an optional feature. This file could be added to the project if the flag is true imo 🙆‍♂️

Comment (luongvo) [reply codecov.yml:3]: Okay, let me remove the CodeCov integration out of this PR and put the CodeCov integration ticket back to In Dev

Comment (luongvo) [reply codecov.yml]: @sleepylee @manh-t updated

**Review by manh-t — APPROVED** [review]
LGTM

**Review by doannimble — APPROVED** [review]
LGTM 💯

**Review by sleepylee — APPROVED** [review]

**Pull request merged**